sdk/src/test/scala/io/horizen/account/state/ContractInteropCallTest.scala

```diff
@@ -1,87 +1,844 @@
+      "gasUsed": "$failedSubCallGasHexString",
+      "input": "0x$STORAGE_INC_ABI_ID",
+      "error": "$expectedErrorMsg"
+    },
+    {
+      "type" : "CALL",
+      "from": "${NativeTestContract.contractAddress}",
+      "to": "$storageContractAddress",
+      "value" : "0x0",
+      "gas": "${NativeTestContract.SUB_CALLS_GAS_HEX_STRING}",
+      "gasUsed" : "0x1a4",
+      "input" : "0x$STORAGE_INC_ABI_ID",
+      "output" : "0x"
+    },
+    {
+      "type" : "STATICCALL",
+      "from": "${NativeTestContract.contractAddress}",
+      "to": "$storageContractAddress",
+      "gas": "${NativeTestContract.SUB_CALLS_GAS_HEX_STRING}",
+      "gasUsed" : "0xf6",
+      "input" : "0x$STORAGE_RETRIEVE_ABI_ID",
+      "output": "0x${BytesUtils.toHexString(currentExpectedValueBytes)}"
+    } ]
+  }""",
+      traceResult
+    )
+
   }
 
   @Test
   def testEvmContractCallingNativeContract(): Unit = {
-    // TODO
+    val nativeCallerAddress = deploy(ContractInteropTestBase.nativeCallerContractCode)
+
+
+    ///////////////////////////////////////////////////////
+    // Test 1: Solidity contract executes a staticcall on a Native Smart Contract, before reaching the fork point.
+    // It should fail because the EVM tries to execute the "fake" code associated with the native smart contract
+    ///////////////////////////////////////////////////////
+    val retrieveInput = BytesUtils.fromHexString(NATIVE_CALLER_STATIC_READONLY_ABI_ID)
+
+    var tracer = new Tracer(new TraceOptions(false, false, false, false, "callTracer", null))
+    val blockContextForFork =
+      new BlockContext(Address.ZERO, 0, FeeUtils.INITIAL_BASE_FEE, gasLimit, 1, 1, 1, 1234, null, Hash.ZERO)
+    blockContextForFork.setTracer(tracer)
+
+    Try(transition(getMessage(nativeCallerAddress, data = retrieveInput), blockContextForFork)) match {
+      case Failure(ex: ExecutionRevertedException) => //OK
+      case res => fail(s"Wrong result: $res")
+    }
+
+    var traceResult = tracer.getResult.result
+    // check tracer output
+
+    println(traceResult)
+    assertJsonEquals(
+      s"""{
+                    "type": "CALL",
+                    "from": "$origin",
+                    "to": "${nativeCallerAddress}",
+                    "gas": "$gasLimitHexString",
+                    "gasUsed": "0x6eb6",
+                    "input": "0x${BytesUtils.toHexString(retrieveInput)}",
+                    "value": "0x0",
+                    "error": "execution reverted",
+                    "calls": [{
+                      "type": "STATICCALL",
+                      "from": "$nativeCallerAddress",
+                      "to": "${NativeTestContract.contractAddress}",
+                      "gas": "${NativeTestContract.SUB_CALLS_GAS_HEX_STRING}",
+                      "gasUsed": "${NativeTestContract.SUB_CALLS_GAS_HEX_STRING}",
+                      "input": "0x$NATIVE_CONTRACT_RETRIEVE_ABI_ID",
+                      "error": "invalid opcode: opcode 0xce not defined"
```

Comment:
Maybe we could add a comment on why we get this 0xce value. As you explained to me it is a value got from EVM which is trying to apply the call internally. The value depends on this particular UT.

Reply:
Done.